src/Registry/TypeRegistry.php

```diff
@@ -883,17 +885,45 @@ protected function prepare_field( $field_name, $field_config, $type_name ) {
 			return null;
 		}
 
+		if ( is_string( $field_config['type'] ) ) {
+			$field_config['type'] = $this->format_key( $field_config['type'] );
+		}
+
 		/**
 		 * If type is not already a callable, create one to preserve lazy-loading.
 		 */
 		if ( ! is_callable( $field_config['type'] ) ) {
-			$field_config['type'] = function () use ( $field_config ) {
-				if ( is_string( $field_config['type'] ) ) {
-					return $this->get_type( $field_config['type'] );
+
+			if ( is_string( $field_config['type'] ) ) {
+				if ( false !== strpos( $field_config['type'], 'graphql_type_' ) ) {
+					if ( $this->has_type( $field_config['type'] ) ) {
+
+						$field_config['type'] = function () use ( $field_config ) {
+							if ( is_string( $field_config['type'] ) ) {
+								return $this->get_type( $field_config['type'] );
+							}
+
+							return $this->setup_type_modifiers( $field_config['type'] );
+						};
+					} else {
```

Comment:
Repeating a ton of logic here. I think we could move it into `get_type` and dry this up quite a bit. Happy to take a crack at it if you want.

Reply:
Have at it. I was getting to a point where my brain was fried and I was almost forgetting what I was trying to solve. lol